Rewrite/src/Rewrite.Xml/XmlVisitor.cs

```diff
@@ -26,15 +26,15 @@ public override bool IsAcceptable(SourceFile sourceFile, P p)
     {
         prolog = prolog.WithMarkers(VisitMarkers(prolog.Markers, p));
         prolog = prolog.WithXmlDecl(VisitAndCast<Xml.XmlDecl>(prolog.XmlDecl, p));
-        prolog = prolog.WithMisc(ListUtils.Map(prolog.Misc, el => (Misc?)Visit(el, p)));
-        prolog = prolog.WithJspDirectives(ListUtils.Map(prolog.JspDirectives, el => (Xml.JspDirective?)Visit(el, p)));
+        prolog = prolog.WithMisc(prolog.Misc.Map(el => (Misc?)Visit(el, p)));
+        prolog = prolog.WithJspDirectives(prolog.JspDirectives.Map(el => (Xml.JspDirective?)Visit(el, p)));
```

Comment:
I only saw you change the use of `ListUtils.Map()` calls in a few other places.

Since this class (and all other visitors) are generated, we would need some kind of consistent pattern we can use. Also, the thing regarding referential equality on the list when the mapping function doesn't change any elements is really important to preserve.

Reply:
I didn't change behavior of that method beyond allowing it to be used as extension method and ensuring it doesn't allow nulls to be passed in (since you were returning it as such anyways but return type was non-nullable). [See change. ]( https://github.com/openrewrite/rewrite-csharp/pull/10/files#diff-690a6da6ecf671993fb60308fe15782166330ec7b143920a01431ea95c94b28a). While the method itself doesn't allow change, when used as extension method we can still call it with `?.` syntax to skip it's invocation altogether and propagate null forward at the call site, allowing us to maintain null checking rules. The extension method syntax should be used across the board as it's more concise and helps solve this null check issue. We will need to modify the generator